File: 0_liste&dict/Py/lc3502.py
from typing import List

class Solution:
    def minCosts(self, cost: List[int]) -> List[int]:
        res = []
        res.append(cost[0])
        
        for i in range(1, len(cost)):
            res.append(min(res[i-1], cost[i]))
        
        return res

File: 0_liste&dict/C++/lc3502.cpp
#include <vector>
#include <algorithm> // Pour std::min

class Solution {
public:
    vector<int> minCosts(vector<int>& cost) {        
        vector<int> res;
        res.push_back(cost[0]);

        for (int i = 1; i < cost.size(); i++) { 
            res.push_back(min(res[i-1], cost[i])); 
        }
        return res;
    }
};